WMF Framework/Event Platform/EventPlatformClient.swift

```diff
@@ -383,26 +353,56 @@ public class EventPlatformClient: NSObject, SamplingControllerDelegate {
      * fire-and-forget fashion
      */
     private func postAllScheduled(_ completion: (() -> Void)? = nil) {
-        DDLogDebug("EPC: Posting all scheduled requests")
+        guard let storageManager = self.storageManager else {
+            return
+        }
+
+        DDLogDebug("EPC: Processing all scheduled requests")
+        let records = storageManager.popAll()
+        if records.count == 0 {
+            DDLogDebug("EPC: Nothing to send.")
+            return
+        }
+
         let group = DispatchGroup()
-        while let item = outputBufferPopFirst() {
+        for record in records {
             group.enter()
-            httpPost(url: item.url, body: item.data) { result in
+
+            guard let stream = Stream(rawValue: record.stream) else {
+                /// This is an unrecoverable logic error. Mark as failed.
+                DDLogError("EPC: Invalid event stream name '\(record.stream)'")
+                storageManager.markFailed(record: record)
+                continue
```

Comment:
You need to add a `group.leave()` before each of these continues - every group.enter() needs to be balanced with one, and these early continues won't reach the one in the `httpPost` completion block.

Reply:
Done.